include/mbedtls/pk_info.h

```diff
@@ -0,0 +1,274 @@
+/**
+ * \file pk_info.h
+ *
+ * \brief Public Key cryptography abstraction layer: object interface
+ *
+ *  This file contains the info structure interface used by developers to
+ *  provide target-specific implementations of opaque key handling functions
+ *  (called engines in the following).
```

Comment:
In keeping with my comment above, proposed rewording:

> This file defines the interface that pk (public-key cryptography) engines must implement. A pk engine defines how a key pair is represented and how to perform operations on a key. Mbed TLS contains built-in engines implemented in software or, if available on the target platform, with hardware acceleration. In addition, it is possible to define opaque key engines, which forward operation requests to an external cryptoprocessor.

> Applications should not access this interface directly and should instead use the functions declared in pk.h.

@hanno-arm What do you think of this wording?

Reply:
I like the wording. Suggestion for slight modification: 
> This file defines the interface the public-key cryptography engines (PK engines) must implement. A PK engine defines how a public-private key pair is represented and how to perform cryptographic operations with it. Mbed TLS contains built-in PK engines implemented either purely in software or with hardware acceleration support, depending on the target platform. In addition, it is possible to define custom opaque key engines that forward operation requests to cryptographic modules outside Mbed TLS, like external cryptoprocessors or general PKCS#11 tokens.

Reply:
@hanno-arm I like that. Just one minor change: “like” → “such as” in the last sentence.

Reply:
Done